### PR TITLE
List StoryTranslations currently being translated or reviewed by User

### PR DESCRIPTION
**Pull request by jennchenn**
## Notion ticket link
[List StoryTranslations currently being translated or reviewed by User](https://www.notion.so/uwblueprintexecs/List-StoryTranslations-currently-being-translated-or-reviewed-by-User-6d4c73c43a5741979392feb916a475f9)

## Implementation description
* added query to return list of StoryTranslations by User based on whether they are a translator (translator=True) or reviewer (translator=False)

## Steps to test
1. Seed database with at least 2 users
```
$ docker exec -it <db container id> psql -U postgres -d planet-read
$ INSERT INTO users (first_name, last_name, auth_id, role, approved_languages) VALUES ('First', 'Last', 'insert-firebase-uid', 'Admin', '{"english":3}');
```
2. Seed database with stories/story translation entries
```
$ INSERT INTO stories (title, description, youtube_link, level) VALUES ('my story title', 'story description', 'link', 1);
$ INSERT INTO story_translations (story_id, language, stage, translator_id, reviewer_id) VALUES (<id of inserted story>, 'ARABIC', 'START', 1, 2);
```
3. Run query and verify the correct values from Story/StoryTranslation are retrieved (in this case a list of stories where user is the translator should be returned)
```
{
  storyTranslationsByUser(userId: 2, translator: true) {
    storyId
    title
    description
    youtube_link
    level
    storyTranslationId
    language
    stage
    translatorId
    reviewerId
  }
}
```
4. Change translate to `false` as well to verify only stories where the user is a reviewer are returned
```
{
  storyTranslationsByUser(userId: 2, translator: false) {
    storyId
    title
    description
    youtube_link
    level
    storyTranslationId
    language
    stage
    translatorId
    reviewerId
  }
}
```

## What should reviewers focus on?
* query in `app/services/implementations/story_service.py`
*  `StoryTranslationResponseDTO` in `backend/python/app/graphql/types/story_type.py` 
     * right now the query returns the id, title, description and level from the Story table and id, language, stage, translator_id, reviewer_id from the StoryTranslation table--are these the fields we want returned/more/less?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/graphql/types/story_type.py [hunk 1]
```diff
@@ -33,3 +33,15 @@ class StoryRequestDTO(graphene.InputObjectType):
...
+
+class StoryTranslationResponseDTO(graphene.ObjectType):
+    story_id = graphene.Int(required=True)
+    title = graphene.String(required=True)
+    description = graphene.String(required=True)
+    level = graphene.Int(required=True)
+    story_translation_id = graphene.Int(required=True)
+    language = graphene.String(required=True)
+    stage = graphene.Field(StageEnum, required=True)
+    translator_id = graphene.Int(required=True)
```
Comment (gaoxk) [inline backend/python/app/graphql/types/story_type.py:46]: `translator_id ` may be null if a translator dropped the story (rare edgecase, we will have to build a way for the translation to be picked up again) 
`reviewer_id ` will be null if the story is in START or TRANSLATION stage 
TLDR these fields are not required

File: backend/python/app/services/implementations/story_service.py [hunk 2]
```diff
@@ -51,3 +52,37 @@ def create_story(self, story, content):
...
+                    Story.description.label("description"),
+                    Story.level.label("level"),
+                    StoryTranslation.id.label("story_translation_id"),
+                    StoryTranslation.language.label("language"),
+                    StoryTranslation.stage.label("stage"),
+                    StoryTranslation.translator_id.label("translator_id"),
+                    StoryTranslation.reviewer_id.label("reviewer_id"),
+                )
+                .join(StoryTranslation, Story.id == StoryTranslation.story_id)
+                .filter(StoryTranslation.reviewer_id == user_id)
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:87]: You could throw in a ternary operator into the filter to avoid repeat code, eg
`.filter(StoryTranslation.translator_id== user_id if translate else StoryTranslation.reviewer_id == user_id)`

File: backend/python/app/services/implementations/story_service.py [hunk 3]
```diff
@@ -51,3 +52,37 @@ def create_story(self, story, content):
         db.session.refresh(new_story)
 
         return new_story
+
+    def get_story_translations(self, user_id, translate):
+        if translate:
+            return (
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:58]: can you wrap this in a try-except and log exceptions? (see `get_user_by_id `or a similar service fn)

File: backend/python/app/graphql/schema.py [hunk 4]
```diff
@@ -26,6 +30,11 @@ class Query(graphene.ObjectType):
     entities = graphene.Field(graphene.List(EntityResponseDTO))
     stories = graphene.Field(graphene.List(StoryResponseDTO))
     story_by_id = graphene.Field(StoryResponseDTO, id=graphene.Int())
+    story_translations_by_user = graphene.Field(
+        graphene.List(StoryTranslationResponseDTO),
+        user_id=graphene.Int(),
+        translate=graphene.Boolean(),
```
Comment (gaoxk) [inline backend/python/app/graphql/schema.py:36]: nit: can we make this param `translator` instead of `translate`? Now that I think about it, I think `translator `makes more sense as we are describing what the user is, rather than their intentions

File: backend/python/app/services/interfaces/story_service.py [hunk 5]
```diff
@@ -38,3 +38,15 @@ def create_story(self, story, content):
         :raises Exception: if entity fields are invalid
         """
         pass
+
+    @abstractmethod
+    def get_story_translations(self, user_id, translate):
+        """Return a list of stories currently being translated/reviewed
+
+        :param user_id: id of the user
+        :param translate: boolean; if True will return a list of stories
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:47]: nit: repeat comment about `translator`

File: backend/python/app/services/interfaces/story_service.py [hunk 6]
```diff
@@ -38,3 +38,15 @@ def create_story(self, story, content):
...
+        being translated, else a list of stories being reviewed
+        :return: list of Story objects
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:49]: nit: its more like the function is returning some story & story translation information together, feel free to directly reference  StoryTranslationResponseDTO

File: backend/python/app/graphql/types/story_type.py [hunk 7]
```diff
@@ -33,3 +33,15 @@ class StoryRequestDTO(graphene.InputObjectType):
     youtube_link = graphene.String(required=True)
     level = graphene.Int(required=True)
     translated_languages = graphene.List(LanguageEnum)
+
+
+class StoryTranslationResponseDTO(graphene.ObjectType):
```
Comment (gaoxk) [inline backend/python/app/graphql/types/story_type.py:38]: We should also return the `youtube_link ` as it's necessary for the story card

**Review by gaoxk — CHANGES_REQUESTED** [review]
looking good so far!

**Review by gaoxk — APPROVED** [review]
🚢 🚢 🚢
